staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -120,6 +120,33 @@ type Config struct {
 	// Version string
 }
 
+// String implements fmt.Stringer and sanitizes sensitive fields of Config to
+// prevent accidental leaking via logs.
+//
+// If you need to print out raw Config for debugging, use %#v.
+func (c *Config) String() string {
+	if c == nil {
+		return "<nil>"
+	}
+	cc := CopyConfig(c)
+	// Explicitly mark non-empty credential fields as redacted.
+	if cc.Username != "" {
```

Comment:
Username is actually pretty useful for local debugging

Reply:
Reverted